consult.el

```diff
@@ -932,6 +936,70 @@ Depending on the selected item OPEN-BUFFER, OPEN-FILE or OPEN-BOOKMARK will be u
   (interactive)
   (consult--buffer #'switch-to-buffer #'find-file #'bookmark-jump))
 
+;;;###autoload
+(defun consult-kmacro (arg)
+  "Run a chosen keyboard macro.  With prefix ARG, run the macro that many times.
+
+Macros containing mouse clicks aren't displayed."
+  (interactive "p")
+  (if (not (or last-kbd-macro kmacro-ring))
```

Comment:
This should be a `unless` and then do not indent the let* since user-error throws. This is what I meant earlier. I prefer if the indentation level is not as deep.